ui/pages/setup_secure.tsx

```diff
@@ -41,32 +41,36 @@ const SetupSecure: NextPage = () => {
       return;
     }
 
+    const returnTo = _returnTo
+      ? _returnTo.toString()
+      : window.location.pathname + `/../setup_complete`;
```

Comment:
This is hacky, @edlerd can we refactor this?

Reply:
Why not define a `const setupComplete = window.location.pathname + `/../setup_complete`;` and keep the previous line 47 below a bit more as it was: `returnTo: returnTo ? String(returnTo) : setupComplete`?

Reply:
I removed the extra variable, thanks for the catch.

But my question still remains about how we should construct the path to the `setup_complete` page.

Reply:
Doesn't the relative path as `./setup_complete` should work just fine?

Reply:
No, the current path will be something like `http://login-ui.com/ui/setup_secure`, this means that the relative path of `./setup_complete` will result to `http://login-ui.com/ui/setup_secure/setup_complete`

The current code just happened to work, because certain parts of the code were unreachable. 
For example:
- https://github.com/canonical/identity-platform-login-ui/pull/287/files#diff-2e493fbf6fdb5973f71ece440a2849e4b4765ab23f84cfdd4a9186c8746f0cc9L87 would never be executed because the condition:
```
if (flow?.state === "success")
```
would never be true because `setFlow` was never called to update the flow with the result of `updateSettingsFlow`.
- The same is true for https://github.com/canonical/identity-platform-login-ui/pull/287/files#diff-2e493fbf6fdb5973f71ece440a2849e4b4765ab23f84cfdd4a9186c8746f0cc9L93-L94, the condition:
```
if (flow?.return_to)
```
was never true
- Also I think that [this](https://github.com/canonical/identity-platform-login-ui/pull/287/files#diff-2e493fbf6fdb5973f71ece440a2849e4b4765ab23f84cfdd4a9186c8746f0cc9L50) was always true, as the flow always has a `request_url`, which means that the `setFlow` at https://github.com/canonical/identity-platform-login-ui/pull/287/files#diff-2e493fbf6fdb5973f71ece440a2849e4b4765ab23f84cfdd4a9186c8746f0cc9L57 was unreachable.

Reply:
> No, the current path will be something like `http://login-ui.com/ui/setup_secure`, this means that the relative path of `./setup_complete` will result to `http://login-ui.com/ui/setup_secure/setup_complete`

That is surprising. Is the current url maybe `http://login-ui.com/ui/setup_secure/` with a trailing slash? Then we should remove that trailing slash and the redirect with the relative path should work fine.

Another option is we could traverse the path upwards until we find `/ui/` and treat that as the root. But this only works under the assumption the ui will always be hosted under a `/ui` prefix. I'd prefer the 1st option as it seems easier to me.

Reply:
But this is not a redirect, this is concatenation:
```
"http://login-ui.com/ui/setup_secure/./setup_complete" = "http://login-ui.com/ui/setup_secure/setup_complete" 
```

And:
```
"http://login-ui.com/ui/setup_secure./setup_complete" = Invalid
```

We construct a URL and give it to Kratos and tell to Kratos that the user should be redirected there AFTER they finish their flow. You don't know at which URL the user will be when they finish their flow (so a relative path does not make much sense).

In your example we would have to tell Kratos to redirect the user to a relative path, but I am not sure if this works. I am pretty sure Kratos won't allow you to set a relative path (e.g. `./setup_complete`) as the return_to URL.

Reply:
I understand what you meant above now. This is not a redirect, but a URL to be used later as we initiate the kratos flow.

Is the path guaranteed to include the `/ui/` prefix? If yes, we could parse the window.location.pathname and remove anything after the last occurrence of `/ui/` to get a base URL. Then we can append `setup_complete` to the base URL.

Reply:
Actually this is way simpler, we are on the `setup_secure` page module here, so just replace `setup_secure` with `setup_secure` in the current pathname should work well.

Reply:
Thanks, I updated it. I don't love that I have to hard-code the file name in the code, but it works